### PR TITLE
vscode bridge to k8s code.

### PR DESCRIPTION
**Pull request by Tatsinnit**
This PR enables the seperation of `vscode extension code of Bridge to kubernetes` from the https://github.com/Azure/Bridge-To-Kubernetes to its own self contained `vscode repository` to achieve:

* Cleaner `vscode release and handling of code`. Original authors will be invited to take a look and see how we have ported this and we can baseline this for future development.
* Quicker turnaround with GH Publish workflows.
* End-to-end extension code under one repo.
* Other advantage for extension which will follow as the repo will grow and community support will happen.

This code was pre existing and removed from parent repo because the other repo had whole lot of cli and vscode merge into one: https://github.com/Azure/Bridge-To-Kubernetes/pull/10 

Hence it made sense to keep vscode repo separately to easy to maintain and for original authors to comment before we merge this.

* We need to rejig the ReadMe.
* Ask few questions from Original Author to get more context. et. al. (before we merge this) 

Thank you 🎉❤️🙏 and cc: @elenavillamil , @hsubramanianaks , @gambtho , @daniv-msft, @sabbour, @qpetraroia . (Please feel free to add more)

Thanks heaps,
^Tats

### CODE REVIEW

File: workflows/publish.yml
Comment (Tatsinnit) [inline workflows/publish.yml:21]: ☕️🪄 this is publish workflow for marketplace. Achieves what is needed for pre publish checks.

File: workflows/build.yml
Comment (Tatsinnit) [inline workflows/build.yml:1]: ☕️💡 this is build yaml for pre flight PR checks when a PR against this repo will be open.

**Comment by Tatsinnit** [comment]
💡 📓 - Read me needs updation, current idea is to reuse the `Bridge to Kubernetes` readme for vscode and update the readme accordingly, here is the link: https://github.com/Azure/Bridge-To-Kubernetes so we can just link the https://docs.microsoft.com/en-us/visualstudio/bridge/bridge-to-kubernetes-vs-code 

cc fort thoughts: @sabbour and @qpetraroia

**Comment by Tatsinnit** [comment]
💡 📓 - Once this PR is merged we will open separate PR for quick Docs addition for release cadence etc. Thanks heaps guys.

**Comment by elenavillamil** [comment]
> Do we want to update readme in this PR? This PR will be open for a few days anyway, so it may make sense to close that item with this PR as well?

**Comment by Tatsinnit** [comment]
> Do we want to update readme in this PR? This PR will be open for a few days anyway, so it may make sense to close that item with this PR as well?

Sounds good to me, I have pushed the ReadMe changes, should we wait for Dan to take e look just as fyi? (eyeball it in case he notices anything) just some thoughts. ☕️🎉💡

File: README.md
Comment (Tatsinnit) [inline README.md:17]: 💡 📓 Telemetry declaration for this repo code here. Thank you @lolodi ❤️🙏☕️

File: package.json
Comment (Tatsinnit) [inline package.json:250]: 💡 How this `$Expected_CLI_Version$` gets passed for this meta information?

Comment (hsubramanianaks) [reply package.json:250]: this is passed via the CI/CD build, we are planning to change this when we open source this.

Comment (Tatsinnit) [reply package.json:250]: Sounds good, this is good for this repo to lock it to latest stable release, thank you. 📓 => version I will update this to is `1.0.20220816.1`.

Comment (hsubramanianaks) [reply package.json:250]: it is 1.0.20220816.2 please refer this blob link to get latest CLI version - https://bridgetokubernetes.azureedge.net/zip/lks.json but this would change once OSS release pipelines are used by vscode changes. we need to change the way vscode download them from github releases instead of blob storage. A new feature for vscode.

Comment (Tatsinnit) [reply package.json:250]: 💡 ok latest reside here - https://bridgetokubernetes.azureedge.net/zip/lks.json verion: `1.0.20220816.2` , I will update this place and then this PR is ready to be merged as it is, because it has old code - just need original authors go-ahead. Thanks.

File: .github/workflows/publish.yml
Comment (daniv-msft) [inline .github/workflows/publish.yml:43]: I don't know if this is actually a potential problem or not, but at this point would this HaaLeo extension have access to the VS_MARKETPLACE_TOKEN? Are there any risks in using an external action here?

File: README.md
Comment (daniv-msft) [inline README.md:6]: Nit: typo, should it be `VS Code`?

Comment (daniv-msft) [reply README.md:6]: Or `Bridge to Kubernetes VS Code extension`? As you prefer

File: README.md
Comment (daniv-msft) [inline README.md:12]: Could you please use the aka.ms link instead? https://aka.ms/bridge-how-it-works

File: README.md
Comment (daniv-msft) [inline README.md:13]: Similar comment: could you please use this link instead: https://aka.ms/bridge-to-k8s-vsc-quickstart

File: README.md
Comment (daniv-msft) [inline README.md:17]: Nit: Do we want to mention `Azure Kubernetes Service` specifically here, given that Bridge supports all Kubernetes clusters?

File: package.json
Comment (daniv-msft) [inline package.json:10]: Should we include a MIT license in the repo? I'm not sure if this is required in this case.

**Review by daniv-msft — APPROVED** [review]

File: .github/workflows/publish.yml
Comment (Tatsinnit) [reply .github/workflows/publish.yml:43]: 💡☕️ Thank for mention and its good to archive this for any future readers, this is just a GH action for vscode publish which uses a GH Secret Variable which cannot be logged.

reference: https://github.com/marketplace/actions/publish-vs-code-extension and for GH secret variable details here: https://docs.github.com/en/actions/security-guides/encrypted-secrets

File: package.json
Comment (Tatsinnit) [reply package.json:10]: Thank you for this mention by default the whole repo is, please let me know if there is anything specific we can change. 🙏☕️

Screen Shot 2022-08-24 at 9 34 48 PM

**Review by hsubramanianaks — APPROVED** [review]

File: package.json
Comment (daniv-msft) [reply package.json:10]: Ah OK didn't see that. Thanks!

**Review by daniv-msft — APPROVED** [review]

**Pull request merged**
